cpp/src/slice2cs/Gen.cpp

```diff
@@ -3117,81 +3084,38 @@ Slice::Gen::TypesVisitor::visitDataMember(const DataMemberPtr& p)
     emitDeprecate(p, cont, _out, "member");
 
     string type = typeToString(p->type(), ns, isOptional);
-    string propertyName = fixId(p->name(), baseTypes, isClass);
-    string dataMemberName;
+    string dataMemberName = fixId(p->name(), baseTypes, isClass);
 
-    if(isProperty)
-    {
-        dataMemberName = "_" + p->name();
-    }
-    else
+    emitAttributes(p);
+    emitGeneratedCodeAttribute();
+    if(isPrivate)
     {
-        dataMemberName = propertyName;
+        _out << nl << "private";
     }
-
-    if(isProperty)
+    else if(isProtected)
     {
-        _out << nl << "private";
+        _out << nl << "protected";
     }
     else
     {
-        emitAttributes(p);
-        emitGeneratedCodeAttribute();
-        if(isPrivate)
-        {
-            _out << nl << "private";
-        }
-        else if(isProtected)
-        {
-            _out << nl << "protected";
-        }
-        else
-        {
-            _out << nl << "public";
-        }
+        _out << nl << "public";
     }
 
-    if(isOptional && isValue)
```

Comment:
This condition is always false: if the data member is optional, it's within a class or an exception, and isValue means the enclosing Slice type is a struct mapped to a C# value type.

If a struct could have optional field (like in Slice 2), this code would be odd and typically incorrect.